### PR TITLE
added delete incident feature

### PR DESCRIPTION
**Pull request by navinkumar-k-015**
Created icon button on incident page when clicked asks confirmation of deletion then delete the incident

### CODE REVIEW

File: lib/clients/incidents_client.dart [hunk 1]
```diff
@@ -12,11 +12,19 @@ class IncidentsClient extends HTTPClient {
     return responseJSON.map((p) => new Incident.fromJson(p)).toList();
   }
 
+
   Future<List<Incident>> getIncidents() async {
     String pageID = await this.getRequestPageId();
-    List responseJSON = await this.requestGet('/pages/$pageID/incidents');
+    List responseJSON =
+        await this.requestGet('/pages/$pageID/incidents?limit=5');
```
Comment (valtlfelipe) [inline lib/clients/incidents_client.dart:19]: I don't think this should be changed

File: lib/pages/incidents_list.dart [hunk 2]
```diff
@@ -77,6 +77,7 @@ class _IncidentsListPageState extends State<IncidentsListPage>
       return null;
     }
   }
+
```
Comment (valtlfelipe) [inline lib/pages/incidents_list.dart:80]: Could you please revert this change too

File: lib/pages/incident.dart [hunk 3]
```diff
@@ -106,6 +116,11 @@ class _IncidentPageState extends State<IncidentPage> {
                   }).toList();
                 },
               ),
+              IconButton(
+                  icon: Icon(Icons.delete),
+                  onPressed: () {
+                    showAlertDialog(context);
```
Comment (valtlfelipe) [inline lib/pages/incident.dart:122]:
```suggestion
                    showRemoveDialog(context);
```

File: lib/pages/incident.dart [hunk 4]
```diff
@@ -200,4 +215,40 @@ class _IncidentPageState extends State<IncidentPage> {
       }).toList(),
     );
   }
+
+  Widget showAlertDialog(BuildContext context) {
```
Comment (valtlfelipe) [inline lib/pages/incident.dart:219]:
```suggestion
  Widget showRemoveDialog(BuildContext context) {
```

File: lib/pages/incident.dart [hunk 5]
```diff
@@ -200,4 +215,40 @@ class _IncidentPageState extends State<IncidentPage> {
...
+    // set up the buttons
+    Widget cancelButton = FlatButton(
+      child: Text("Cancel"),
+      onPressed: () {
+        Navigator.pop(context);
+      },
+    );
+    Widget deleteButton = FlatButton(
+      child: Text("Delete"),
```
Comment (valtlfelipe) [inline lib/pages/incident.dart:228]:
```suggestion
      child: Text("Delete", style: TextStyle(color: Colors.red)),
```

File: lib/pages/incident.dart [hunk 6]
```diff
@@ -200,4 +215,40 @@ class _IncidentPageState extends State<IncidentPage> {
...
+        Navigator.of(context)
+            .push(MaterialPageRoute(builder: (BuildContext context) {
+          return IncidentsListPage();
+        }));
+      },
+    );
+    // set up the AlertDialog
+    AlertDialog alert = AlertDialog(
+      title: Text("AlertDialog"),
+      content: Text("Are you sure you like to delete this incident"),
```
Comment (valtlfelipe) [inline lib/pages/incident.dart:240]:
```suggestion
      title: Text("Are you sure?"),
      content: Text("This cannot be undone and will remove all associated data."),
```

File: lib/pages/incident.dart [hunk 7]
```diff
@@ -200,4 +215,40 @@ class _IncidentPageState extends State<IncidentPage> {
...
+      },
+    );
+    Widget deleteButton = FlatButton(
+      child: Text("Delete"),
+      onPressed: () {
+        _delete();
+        Navigator.of(context)
+            .push(MaterialPageRoute(builder: (BuildContext context) {
+          return IncidentsListPage();
+        }));
```
Comment (valtlfelipe) [inline lib/pages/incident.dart:234]: This will push a new page to the stack, on going back it would go back to the incident that was removed. I think this should work
```suggestion
        Navigator.of(context).popUntil(ModalRoute.withName("/home"));
```
And then add this to the home route definition inside `main.dart`
```
MaterialPageRoute(
	settings: RouteSettings(name: "/home"), // this is the line to add
	...);
```

**Review by valtlfelipe — COMMENTED** [review]
Awesome! Thanks for your PR!
I left some comments/suggestions, feel free to to ask anything.

File: lib/clients/incidents_client.dart (hunk 1 above)
Comment (navinkumar-k-015) [reply lib/clients/incidents_client.dart]: Yeah sorry for that, I was trying for pagination then I forgot to remove.

**Comment by navinkumar-k-015** [comment]
I had pushed all the changes you suggested and thanks for giving suggestions.

File: lib/clients/incidents_client.dart (hunk 1 above)
Comment (valtlfelipe) [reply lib/clients/incidents_client.dart]: No worries! :)

**Comment by navinkumar-k-015** [comment]
can you please label it under hacktoberfest-accepted

File: lib/pages/incident.dart [hunk 8]
```diff
@@ -200,4 +215,41 @@ class _IncidentPageState extends State<IncidentPage> {
...
+        style: TextStyle(color: Colors.red),
+      ),
+      onPressed: () {
+        _delete();
+        Navigator.of(context).popUntil(ModalRoute.withName("/home"));
+      },
+    );
+    // set up the AlertDialog
+    AlertDialog alert = AlertDialog(
+      title: Text("Are you sure"),
```
Comment (valtlfelipe) [inline lib/pages/incident.dart:239]: Just missed the question mark
```suggestion
      title: Text("Are you sure?"),
```

**Review by valtlfelipe — COMMENTED** [review]
Just two suggestions, and we are ready to go
(the other one is above)

**Comment by valtlfelipe** [comment]
> can you please label it under hacktoberfest-accepted

I don't have the label set up, but as soon as I approve it, it will count for you 😄 
Don't worry, I will approve it

File: lib/pages/incident.dart (hunk 8 above)
Comment (navinkumar-k-015) [reply lib/pages/incident.dart]: Done!!

File: lib/pages/incidents_list.dart (hunk 2 above)
Comment (navinkumar-k-015) [reply lib/pages/incidents_list.dart:80]: already done it on 3rd commit

Comment (valtlfelipe) [reply lib/pages/incidents_list.dart:80]: Sorry, missed it

**Review by valtlfelipe — APPROVED** [review]
Awesome! You rock! 
Thanks again
